scripts/cat/cats.py

```diff
@@ -819,79 +824,67 @@ def add_to_clan(self) -> list:
                 History.add_beginning(child)
                 ids.append(child_id)
 
         return ids
 
-    def status_change(self, new_status, resort=False):
+    def status_change(self, new_status: StatusEnum | str, resort=False):
         """Changes the status of a cat. Additional functions are needed if you want to make a cat a leader or deputy.
-        new_status = The new status of a cat. Can be 'apprentice', 'medicine cat apprentice', 'warrior'
-                    'medicine cat', 'elder'.
+        new_status = The new status of a cat.
         resort = If sorting type is 'rank', and resort is True, it will resort the cat list. This should
                 only be true for non-timeskip status changes."""
+
+        # auto-converting string
+        if isinstance(new_status, str):
+            new_status = StatusEnum(new_status)
+
         old_status = self.status
         self.status = new_status
         self.name.status = new_status
 
+        if new_status.is_outside_clan():
+            return
+
         self.update_mentor()
         for app in self.apprentice.copy():
             fetched_cat = Cat.fetch_cat(app)
             if isinstance(fetched_cat, Cat):
                 fetched_cat.update_mentor()
 
         # If they have any apprentices, make sure they are still valid:
-        if old_status == "medicine cat":
+        if old_status.is_medcat():
             game.clan.remove_med_cat(self)
 
         # updates mentors
-        if self.status == "apprentice":
+        if new_status.is_app_any():
             pass
 
-        elif self.status == "medicine cat apprentice":
+        elif new_status.is_mediator_any():
             pass
 
-        elif self.status == "warrior":
-            if old_status == "leader":
-                if game.clan.leader:
-                    if game.clan.leader.ID == self.ID:
-                        game.clan.leader = None
-                        game.clan.leader_predecessors += 1
-
-                    # don't remove the check for game.clan, this is needed for tests
-            if game.clan and game.clan.deputy:
-                if game.clan.deputy.ID == self.ID:
-                    game.clan.deputy = None
-                    game.clan.deputy_predecessors += 1
-
-        elif self.status == "medicine cat":
+        elif new_status.is_medcat():
             if game.clan is not None:
                 game.clan.new_medicine_cat(self)
 
-        elif self.status == "elder":
-            if old_status == "leader":
+        elif new_status.is_warrior() or new_status.is_elder():
+            if old_status.is_leader():
                 if game.clan.leader:
                     if game.clan.leader.ID == self.ID:
                         game.clan.leader = None
                         game.clan.leader_predecessors += 1
 
-            if game.clan.deputy:
+            if game.clan and game.clan.deputy:  # don't remove the check for game.clan, this is needed for tests
```

Comment:
I'd reword this to state _why_ it is needed.

Reply:
It's hard to explain it succinctly. Would something like this be clear enough? `game.clan check is needed to let tests work as tests don't have an instance of game`

Reply:
Sounds good to me!